packages/x-components/src/plugins/x-plugin.mixin.ts

```diff
@@ -37,20 +47,27 @@ export const createXComponentAPIMixin = (
  * Creates an object containing the API related to the {@link XBus}.
  *
  * @param bus - The global {@link XBus}.
- * @param xComponent - The root {@link XComponent} that the component that owns this API has.
+ * @param rootComponent - The root {@link XComponent} that the component that owns this API has.
+ * @param origin - The Vue component injected origin.
+ *
  * @returns An object containing the {@link XComponentBusAPI}.
  * @internal
  */
-export function getBusAPI(bus: XBus, xComponent: XComponent | undefined): XComponentBusAPI {
+export function getBusAPI(
+  bus: XBus,
+  rootComponent: XComponent | undefined,
+  origin: QueryOrigin
```

Comment:
Instead of adding a new parameter, because both `rootComponent` and `origin` are properties available in the `this` context, I would just pass the `this`.  This way we have less parameters in the function, and it is already retrieved dynamically, so when we make the change to allow changing it we don't have to remove the code.

As the `this` is now more complex, I would extract it to an interface

```ts
interface PrivateExtendedVueComponent extends Vue {
  xComponent: XComponent | undefined;
  origin: QueryOrigin;
}
```

```ts
): ComponentOptions<Vue> & ThisType<PrivateExtendedVueComponent> => ({
  inject: {
    origin: {
      default: undefined
    }
  },
  created(): void {
    this.xComponent = getRootXComponent(this);

    const aliasAPI = getAliasAPI(this.$store);
    const busAPI = getBusAPI(bus, this);

    this.$x = Object.assign(aliasAPI, busAPI);
  }
});
```

```ts
    emit: <Event extends XEvent>(
      event: Event,
      payload?: XEventPayload<Event>,
      metadata: Omit<WireMetadata, 'moduleName' | 'origin'> = {}
    ) => {
      const rootComponent = component.xComponent;
      const moduleName = rootComponent ? getXComponentXModuleName(rootComponent) : null;
      bus.emit(event, payload as any, { moduleName, origin: component.origin, ...metadata });
      rootComponent?.$emit(event, payload);
    },
    on: bus.on.bind(bus)
```

Another question is, should we try to block passing the origin to the `$x.emit` function like we do with the module name? Or do we want to leave this option open?

Reply:
I've discussed about it with Beltrán and Guille while we were working in the other related task and we should block it.

Reply:
what do you mean with block?
`metadata: Omit<WireMetadata, 'moduleName' | 'origin'> = {}` this in the emit type?

Reply:
Yes, it's already done